service/raft/raft_rpc.cc

```diff
@@ -25,140 +25,109 @@ raft_ticker_type::time_point timeout() {
     return raft_ticker_type::clock::now() + raft_tick_interval * (raft::ELECTION_TIMEOUT.count() / 2);
 }
 
+void log_missing_raft_address(raft::server_id srv_id,
+        const std::experimental::source_location location = std::experimental::source_location::current()) {
+    rlogger.debug("{}:{}: {} dropping outgoing message to {} - IP address not found",
+        location.file_name(), location.line(), location.function_name(), srv_id);
+}
+
 raft_rpc::raft_rpc(raft_state_machine& sm, netw::messaging_service& ms,
         raft_address_map& address_map, raft::group_id gid, raft::server_id srv_id)
     : _sm(sm), _group_id(std::move(gid)), _server_id(srv_id), _messaging(ms), _address_map(address_map)
 {}
 
-future<raft::snapshot_reply> raft_rpc::send_snapshot(raft::server_id id, const raft::install_snapshot& snap, seastar::abort_source& as) {
-    return ser::raft_rpc_verbs::send_raft_send_snapshot(&_messaging,
-        netw::msg_addr(_address_map.get_inet_address(id)), db::no_timeout, _group_id, _server_id, id, snap);
-}
-
-future<> raft_rpc::send_append_entries(raft::server_id id, const raft::append_request& append_request) {
-    return ser::raft_rpc_verbs::send_raft_append_entries(&_messaging,
-        netw::msg_addr(_address_map.get_inet_address(id)), db::no_timeout, _group_id, _server_id, id, append_request);
-}
 
-void raft_rpc::send_append_entries_reply(raft::server_id id, const raft::append_reply& reply) {
-    (void)with_gate(_shutdown_gate, [this, id, &reply] {
-        return ser::raft_rpc_verbs::send_raft_append_entries_reply(&_messaging, netw::msg_addr(_address_map.get_inet_address(id)), timeout(), _group_id, _server_id, id, reply)
-            .handle_exception([id] (std::exception_ptr ex) {
+template <typename Verb, typename Msg> void
+raft_rpc::one_way_rpc(const std::string rpc_name, raft::server_id id,
```

Comment:
There's a lot of copying of `rpc_name`, which will cause unnecessary allocations.

I think you should remove `const` and `std::move(rpc_name)` when capturing it in the lambda (i.e. `[..., rpc_name = std::move(rpc_name), ...]`). Note that the outer lambda must be made `mutable` in order for the `move` in the innter lambda (the one in `handle_exception`) to make a difference.

Reply:
In fact, we should not need `rpc_name` at all. We need to pass `source_location` here anyway (see my other comment), then we could use that for getting the caller's name.

Reply:
I agree and I'll drop rpc_name, will use source location instead.

Reply:
Fixed.